### ISSUE TITLE
Questions about gaussian params when tracking

### ISSUE THREAD

**Issue by Buffyqsf** [spla-tam/SplaTAM#75]
As the code below in utils.slam_helpers, when setting gaussians_grad false, only params['means3D'] are detached, but other params about gaussians can still be optimized? If so, it's not consistent with what's said in the paper
![image](https://github.com/spla-tam/SplaTAM/assets/89965829/d16636b6-ede7-441d-bf9f-e10563a6a177)

![image](https://github.com/spla-tam/SplaTAM/assets/89965829/d393c414-e408-490d-8293-2e89c559edb9)

**Comment by jeezrick** [comment]
It's about the rationale of autograd. When calculating transformed_pts, it only used cam_rot, cam_trans and means3D. So, you only need to decide if you want to involve these three variables into the grad tracking process. 

Like when you do tracking, when backward the grad, the grad will flow from transformed_pts to (cam_rot, cam_trans and means3D), the other Gaussian parameters like color and opacity wasn't even invovled, so they won't get optimized, as they are't part of the compute.

**Comment by Buffyqsf** [comment]
> It's about the rationale of autograd. When calculating transformed_pts, it only used cam_rot, cam_trans and means3D. So, you only need to decide if you want to involve these three variables into the grad tracking process.
> 
> Like when you do tracking, when backward the grad, the grad will flow from transformed_pts to (cam_rot, cam_trans and means3D), the other Gaussian parameters like color and opacity wasn't even invovled, so they won't get optimized, as they are't part of the compute.

Thanks for your answering! I think I know the answer, but it's not like what you said(wasn't invovled). I think the reason is that we set learning rate to zero like below. 
![image](https://github.com/spla-tam/SplaTAM/assets/89965829/0875ca34-6c78-4c32-8321-93995e72bf17)

**Comment by jeezrick** [comment]
> > It's about the rationale of autograd. When calculating transformed_pts, it only used cam_rot, cam_trans and means3D. So, you only need to decide if you want to involve these three variables into the grad tracking process.
> > Like when you do tracking, when backward the grad, the grad will flow from transformed_pts to (cam_rot, cam_trans and means3D), the other Gaussian parameters like color and opacity wasn't even invovled, so they won't get optimized, as they are't part of the compute.
> 
> Thanks for your answering! I think I know the answer, but it's not like what you said(wasn't invovled). I think the reason is that we set learning rate to zero like below. ![image](https://private-user-images.githubusercontent.com/89965829/307202900-0875ca34-6c78-4c32-8321-93995e72bf17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2NzQ1NTAsIm5iZiI6MTcwODY3NDI1MCwicGF0aCI6Ii84OTk2NTgyOS8zMDcyMDI5MDAtMDg3NWNhMzQtNmM3OC00YzMyLTgzMjEtOTM5OTVlNzJiZjE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIzVDA3NDQxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzZjg1YzQ0MjVjNGM5ZDVkYTk1ZDc1NGUxZDQzM2NlMTQzNDdkZjlkZjRkMDFhZTZiNzRkMjA5ZDAyNjFhZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E2WdSoiddtHm31VYa46iOUP5SDxeTsrc2xhN1miLwzI)

I think both reasons are correct. with detach(), means3D won't get any gradients during tracking; Also just set lr to 0, even it has gradient, it won't get updated. 

So, both part make sense. 

Also, if the lrs setting is the whole reason, why they write code like 'params['means3D'].detach()', I think you can verify this by debug the code: you can check that in every iteration in tracking, the gradient if params['means3D'] is None, so even if you set lrs['means3D'] to 0.1, it still work.

So, I guess Im saying that both way works.

**Comment by Nik-V9** [comment]
I agree that this was an oversight on our end, however, the 0 learning rate prevents any bugs :)
